sys/net/gnrc/network_layer/ipv6/nib/_nib-internal.c

```diff
@@ -384,4 +422,22 @@ static inline bool _node_unreachable(_nib_onl_entry_t *node)
     }
 }
 
+uint32_t _evtimer_lookup(const void *ctx, uint16_t type)
+{
+    evtimer_msg_event_t *event = (evtimer_msg_event_t *)_nib_evtimer.events;
+    uint32_t offset = 0;
+
+    DEBUG("nib: lookup ctx = %p, type = %u\n", (void *)ctx, type);
+    while (event != NULL) {
+        offset += event->event.offset;
+        if ((event->msg.type == type) &&
+            ((ctx == NULL) || (event->msg.content.ptr == ctx))) {
+            mutex_unlock(&_nib_mutex);
```

Comment:
There was no previous `mutex_lock` in this function. Please extend the doxygen of this function with a `@pre mutex lock needed` or similar, if this was your intention.